File: 3. Longest Substring without Repeating Chars/solution.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int lswrc = 0;
        for(int index = 0; index < s.length(); ++index){
            Set<Character>currentSubstring = new HashSet<>();
            char currentChar = s.charAt(index);
            int currentIndex = index;
            while(!(currentSubstring.contains(currentChar))){
                currentSubstring.add(currentChar);
                ++currentIndex;
                if(currentIndex>s.length()-1){
                    break;
                }
                currentChar = s.charAt(currentIndex);
            }
            lswrc = Math.max(lswrc, currentSubstring.size());
        }
        return (lswrc);
    }
}

File: 3. Longest Substring without Repeating Chars/solution.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        lswrc = 0
        for index,element in enumerate(s):
            currentSubstring = {}
            currentChar = element
            currentIndex = index
            while(not(currentChar in currentSubstring)):
                currentSubstring[currentChar] = None
                currentIndex+=1
                if(currentIndex==len(s)):
                    break
                currentChar = s[currentIndex]
            lswrc = max(lswrc, len(currentSubstring))
        return lswrc